pkg/datastore/filedb/filter.go

```diff
@@ -15,19 +15,181 @@
 package filedb
 
 import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+	"unicode"
+
 	"github.com/pipe-cd/pipecd/pkg/datastore"
 )
 
-// TODO: Implement filterable interface for each collection.
 type filterable interface {
 	Match(e interface{}, filters []datastore.ListFilter) (bool, error)
 }
 
 func filter(col datastore.Collection, e interface{}, filters []datastore.ListFilter) (bool, error) {
```

Comment:
Return fast when there was no filter to avoid unneeded marshaling.

Reply:
Nice catch 💯

Reply:
Addressed by https://github.com/pipe-cd/pipecd/pull/3389/commits/fad86153a3965ee0083b1ac2c76394d6eafdd36e 🙏